### ISSUE TITLE
Unable to execute code within Cucumber hook specified using @Before and @AfterClass annotations while running Cucumber tests in parallel mode using failsafe plugin

### ISSUE THREAD

**Issue by ruzeynalov** [cucumber/cucumber-jvm#1649]
## Summary

Unable to execute code annotated with @Before and @AfterClass while running Cucumber tests in parallel mode using failsafe plugin

## Expected Behavior

Code within methods annotated with @Before & @AfterClass executed before each scenario and after whole test suite correspondingly

## Current Behavior

Code annotated with @Before and @AfterClass **ignored** while executing cucumber tests in parallel mode with failsafe plugin

When I remove configuration section of fail-safe plugin (threadCount & parallel settings) code annotated with @Before and @After is successfully executed

## Steps to Reproduce (for bugs)
Here is my pom.xml configuration

    
```
<properties>
        <java.version>11</java.version>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
        <compiler.plugin.version>3.8.1</compiler.plugin.version>

        <cucumber.version>4.3.1</cucumber.version>
        <!--Junit-->
        <junit.version>4.12</junit.version>
<!--        <junit.parallelThreadCount>3</junit.parallelThreadCount>-->
<!--        <junit.parallel>all</junit.parallel>-->

    </properties>




<plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-failsafe-plugin</artifactId>
                <version>2.22.2</version>
                <configuration>
<!--                    <threadCount>${junit.parallelThreadCount}</threadCount>-->
<!--                    <parallel>${junit.parallel}</parallel>-->
                    <testFailureIgnore>false</testFailureIgnore>
                    <includes>
                        <include>${run.classes}</include>
                    </includes>
                    <environmentVariables>
                        <app_env>${app_env}</app_env>
                    </environmentVariables>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>integration-test</goal>
                            <goal>verify</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

My Junit class with RunWith configuration

```
@RunWith(Cucumber.class)
@CucumberOptions(
        plugin = {"io.qameta.allure.cucumber4jvm.AllureCucumber4Jvm"},
        junit = {"--filename-compatible-names"},
        tags = {"not @Ignore", "not @InDev"}
)

public class RunAcceptanceTests {
    @AfterClass
    public static void teardown()
    {
             ............my code.....................
    }
}
```

## Environment

* Version used: cucumber 4.3.1, maven 3.6.0, fail-safe plugin 2.22, java 11.0.2
* Operating System and version: Windows 10

**Comment by mpkorstanje** [comment]
I can't reproduce this behavior. Please clone [cucumber-java-skeleton](
https://github.com/cucumber/cucumber-java-skeleton) and provide a executable reproducer. Remove all elements that aren't necessary  (e.g. `includes`, `execution`, `environmentVariables`, ect) to reproduce the problem and inline all properties.

**Comment by mpkorstanje** [comment]
So given that `ParentRunner.run` executes the statement created by `classBlock` and given that `classBlock` handles the both the before and after class hooks, and given that `Cucumber` doesn't override any of these methods I doubt the problem lies within Cucumber or Surefire.

@ruzeynalov I strongly suspect you are being mislead by your output. Try attaching timestamps.

**Comment by ruzeynalov** [comment]
I'm sorry for being whiner :) but how come the issue is reproduced only when I set parallel mode on with corresponding junit annotation?

When I comment configuration block of failsafe plugin, all of my code annotated with @Before and @After (cucumber hooks) successfully executed?

I will prepare the sample for reproduced with skeleton project you've suggested

**Comment by ruzeynalov** [comment]
By the way the issue was resolved after moving from junit 4.12 to junit-vintage 5.50

**Comment by vayaszsolt** [comment]
@ruzeynalov 

Hi.

I am encountering the same scenario as you. Unfortunately if I will move to use junit-vintage then the failsafe plugin will stop to execute the tests in parallel. The tests are executed in parallel with junit4. 
The following are my dependencies:
serenity-screenplay 3.3.2
serenity-screenplay-webdriver  3.3.2
serenity-screenplay-rest  3.3.2
serenity-ensure  3.3.2
serenity-spring  3.3.2
serenity-junit  3.3.2
serenity-cucumber  3.3.2
spring-boot-starter-test 2.7.3
spring-beans 5.3.22
spring-rabbit 2.4.6

The above dependency are executing the tests in parallel using the failsafe plugin. But I remove serenity-junit and replace with serenity-junit5 the tests aren't executed anymore in parallel. 

Do you have an idea where can be the problem? Or other approach? I am searching and debugging for days...
